### PR TITLE
[otbn] Expand the three writeback variants of bn.mulqacc in docs

### LINKED ISSUE #2777: [otbn] Would it make sense to split `bn.mulqacc` into several different instructions?
The current syntax for `mulqacc` is this:
```
      BN.MULQACC [<wb_variant>][<zero_acc>] [<wrd><wrd_hwsel>,] <wrs1>.<wrs1_qwsel>, <wrs2>.<wrs2_qwsel>, <acc_shift_imm>
```

The `<wrd>` argument is required if and only if `<wb_variant>` is specified. This is a bit complicated. Would it make more sense to define one opcode for when writeback is enabled and one for when it's not? (This would not have any effect on the encoding or the hardware; just on assembly syntax).

### PR DESCRIPTION
**Pull request by rswarbrick**
Before this change, an operand was optional or required, depending on
the value of another operand. Now, the writeback variant on which this
depended is encoded in the mnemonic.

We could go further, duplicating these into 6 mnemonics to get rid of
the `<zero_acc>` argument, but I feel like that adds bloat without any
massive advantage.

Closes #2777.

### CODE REVIEW

**Comment by GregAC** [comment]
This LGTM though the documentation ends up placing the operation pseudo-code in the first variant and it's not immediately obvious the second and third variants are also using the same operation pseudo-code, perhaps we could place the operation pseudo-code after all three variants have their fields and decode described?

**Comment by rswarbrick** [comment]
Oops, yes, that was an oversight. (I got as far as adding an `&anchor` where I'd defined it, but didn't `*reference` at the other end). Fixed now.

File: hw/ip/otbn/data/insns.yml [hunk 1]
```diff
@@ -491,6 +473,72 @@ insns:
...
+      - *mulqacc-wrs1
+      - *mulqacc-wrs1-qwsel
+      - *mulqacc-wrs2
+      - *mulqacc-wrs2-qwsel
+      - *mulqacc-acc-shift-imm
+    syntax: |
+      [<zero_acc>] <wrd>, <wrs1>.<wrs1_qwsel>, <wrs2>.<wrs2_qwsel>, <acc_shift_imm>
+    doc: |
+      Multiplies two `WLEN/4` WDR values and adds the result to an accumulator after shifting it.
+      Writes the resulting accumulator to <wrd>.
```
Comment (imphil) [inline hw/ip/otbn/data/insns.yml:493]:
```suggestion
      Writes the resulting accumulator to `<wrd>`.
```

(Otherwise it's a HTML tag and not rendered.)

Not sure what the solution would be (or if we can find an easy one), but it would be potentially nice to get an indication that `doc` requires a different escaping/syntax than the `syntax` section above it.

File: hw/ip/otbn/data/insns.yml [hunk 2]
```diff
@@ -491,6 +473,72 @@ insns:
...
+      - *mulqacc-wrs1-qwsel
+      - *mulqacc-wrs2
+      - *mulqacc-wrs2-qwsel
+      - *mulqacc-acc-shift-imm
+    syntax: |
+      [<zero_acc>] <wrd><wrd_hwsel>,
+      <wrs1>.<wrs1_qwsel>, <wrs2>.<wrs2_qwsel>, <acc_shift_imm>
+    doc: |
+      Multiplies two `WLEN/4` WDR values and adds the result to an accumulator after shifting it.
+      Writes the lower half-word of the resulting accumulator to a `WLEN/2`-bit half-word of <wrd>.
```
Comment (imphil) [inline hw/ip/otbn/data/insns.yml:528]: This line should be something like 
"Shift out the lower half-word of the value stored in the accumulator to a WLEN/2-sized half-word of the destination WDR. The destination half-word is selected by the “wrd_hwsel” field."

File: hw/ip/otbn/data/insns.yml [hunk 3]
```diff
@@ -491,6 +473,72 @@ insns:
...
+      b = UInt(wrs2)
+
+      d_hwsel = None
+      a_qwsel = DecodeQuarterWordSelect(wrs1_qwsel)
+      b_qwsel = DecodeQuarterWordSelect(wrs2_qwsel)
+    operation: *mulqacc-operation
+
+  - mnemonic: bn.mulqacc.so
+    group: bignum
+    synopsis: Quarter-word Multiply and Accumulate with half-word writeback
```
Comment (imphil) [inline hw/ip/otbn/data/insns.yml:509]: Can you rephrase that to be different from the .wo variant, and include "shift" in it somewhere?

File: hw/ip/otbn/data/insns.yml (hunk 1 above)
Comment (rswarbrick) [reply hw/ip/otbn/data/insns.yml:493]: Oops, good point. Fixed. For the general "documentation" of the YAML format, there are some comments about the file format at the top of the file. I'm not sure we need much more: after all, if someone makes a small change to a documentation string, they'll presumably regenerate the HTML and eyeball it to check it did what they expected.

File: hw/ip/otbn/data/insns.yml (hunk 2 above)
Comment (rswarbrick) [reply hw/ip/otbn/data/insns.yml:528]: Hopefully this is a bit clearer now.

**Comment by rswarbrick** [comment]
Force-push fixes HTML tags and rewords things slightly.

@GregAC: Could you take a quick glance to check this matches what you think the instructions do?
